comps/retrievers/langchain/pgvector/retriever_pgvector.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import time
+
+from config import EMBED_MODEL, INDEX_NAME, PG_CONNECTION_STRING
+from langchain_community.embeddings import HuggingFaceBgeEmbeddings, HuggingFaceHubEmbeddings
+from langchain_community.vectorstores import PGVector
+from langsmith import traceable
+
+from comps import (
+    EmbedDoc768,
+    SearchedDoc,
+    ServiceType,
+    TextDoc,
+    opea_microservices,
+    register_microservice,
+    register_statistics,
+    statistics_dict,
+)
+
+tei_embedding_endpoint = os.getenv("TEI_EMBEDDING_ENDPOINT")
+
+
+@register_microservice(
+    name="opea_service@retriever_pgvector",
+    service_type=ServiceType.RETRIEVER,
+    endpoint="/v1/retrieval",
+    host="0.0.0.0",
+    port=7000,
+)
+@traceable(run_type="retriever")
+@register_statistics(names=["opea_service@retriever_pgvector"])
+def retrieve(input: EmbedDoc768) -> SearchedDoc:
+    start = time.time()
```

Comment:
Once we bring in telemetry piece, ideally the trace itself should give us the required statistics. While it is OK to leave this as is for now, I would prefer that we bring in necessary abstraction to ensure migration to a proper telemetry f/w will not require us to change the code all over.

Reply:
sure